src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -44,9 +45,9 @@ public function __construct(ClassMetadataFactoryInterface $classMetadataFactory
 
         $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
 
-        $this->objectClassResolver = $objectClassResolver ?? function ($class) {
-            return \is_object($class) ? \get_class($class) : $class;
-        };
+        $this->objectClassResolver = $objectClassResolver
+            ? $objectClassResolver(...)
+            : static fn ($class) => \is_object($class) ? \get_class($class) : $class;
```

Comment:
now I'm wondering:
>The `fn` keyword indicates a function that will auto-capture variables, by value.
doesn't this mean that by using `fn`, we prevent all variables in the scope from being garbage collected until the closure is? if yes, then we might consider banning `fn` from the codebase...

Reply:
Good news here: short closures only capture the variables that are used in their body.